repose-aggregator/core/repose-core/src/test/groovy/org/openrepose/nodeservice/httpcomponent/RequestProxyServiceImplTest.groovy

```diff
@@ -161,4 +169,188 @@ class RequestProxyServiceImplTest extends Specification {
+                .thenReturn(new BasicHttpResponse(HttpVersion.HTTP_1_1, 204, "OK"))
+
+        when:
+        requestProxyService.put("http://www.google.com", "", Collections.emptyMap(), [] as byte[])
+
+        then:
+        noAssertionError { verify(httpClientService).getClient(null) }
+    }
+
+    def "put(uri, path, headers, body, connPoolId) will try to use a given connection pool"() {
+        given:
+        String testPoolId = "test-pool-id"
+        when(httpClient.execute(any(HttpUriRequest.class)))
+                .thenReturn(new BasicHttpResponse(HttpVersion.HTTP_1_1, 204, "OK"))
+
+        when:
+        requestProxyService.put("http://www.google.com", "", Collections.emptyMap(), [] as byte[], testPoolId)
+
+        then:
+        noAssertionError { verify(httpClientService).getClient(testPoolId) }
+    }
+
+    def "patch(uri, path, headers, body) will try to use a given connection pool"() {
+        given:
+        when(httpClient.execute(any(HttpUriRequest.class)))
+                .thenReturn(new BasicHttpResponse(HttpVersion.HTTP_1_1, 204, "OK"))
+
+        when:
+        requestProxyService.patch("http://www.google.com", "", Collections.emptyMap(), [] as byte[])
+
+        then:
+        noAssertionError { verify(httpClientService).getClient(null) }
+    }
+
+    def "patch(uri, path, headers, body, connPoolId) will try to use a given connection pool"() {
+        given:
+        String testPoolId = "test-pool-id"
+        when(httpClient.execute(any(HttpUriRequest.class)))
+                .thenReturn(new BasicHttpResponse(HttpVersion.HTTP_1_1, 204, "OK"))
+
+        when:
+        requestProxyService.patch("http://www.google.com", "", Collections.emptyMap(), [] as byte[], testPoolId)
+
+        then:
+        noAssertionError { verify(httpClientService).getClient(testPoolId) }
+    }
+
+    def "delete(uri, path, headers) will try to use a given connection pool"() {
+        given:
+        when(httpClient.execute(any(HttpUriRequest.class)))
+                .thenReturn(new BasicHttpResponse(HttpVersion.HTTP_1_1, 204, "OK"))
+
+        when:
+        requestProxyService.patch("http://www.google.com", "", Collections.emptyMap())
+
+        then:
+        noAssertionError { verify(httpClientService).getClient(null) }
+    }
+
+    def "delete(uri, path, headers, connPoolId) will try to use a given connection pool"() {
+        given:
+        String testPoolId = "test-pool-id"
+        when(httpClient.execute(any(HttpUriRequest.class)))
+                .thenReturn(new BasicHttpResponse(HttpVersion.HTTP_1_1, 204, "OK"))
+
+        when:
+        requestProxyService.delete("http://www.google.com", "", Collections.emptyMap(), testPoolId)
+
+        then:
+        noAssertionError { verify(httpClientService).getClient(testPoolId) }
+    }
+
+    // TODO: Use Spock Mocking framework to avoid this hack (caused by Mockito mocks returning null)
+    def noAssertionError(Closure verification) {
```

Comment:
not sure why this was necessary. i haven't run into issues before jsut doing mockito verify's